draco/tests/test_generate.py

```diff
@@ -0,0 +1,27 @@
+from pathlib import Path
+
+from draco import run_clingo
+from draco.programs import constraints, define, generate, hard, helpers
+
+asp_path = Path(__file__).resolve().parent.parent / "asp"
+
+
+def test_generate():
+    prog = (
+        generate.program
+        + define.program
+        + helpers.program
+        + hard.program
+        + constraints.program
+    )
+    with open(asp_path / "examples" / "scatter.lp") as file:
+        scatter = file.readlines()
+    scatter = ("").join(scatter)
+
+    models = list(
+        run_clingo(
+            prog + scatter + ":- {entity(mark,_,_)} < 3.", 10, False, {"max_marks": 3}
```

Comment:
Let's add another unit test (in the run tests) that tests whether you can pass in a constant. 

The program could be

```asp
#const foo.
a(0...foo).
```

And then you can test whether the number of `a`s is `foo` and vary `foo` a bit in the test.